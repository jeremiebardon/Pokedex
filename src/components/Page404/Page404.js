import React from 'react';

export default function Page404() {
  return (
    <div>
      <h3>
        You are in the Void
      </h3>
    </div>
  );
}
