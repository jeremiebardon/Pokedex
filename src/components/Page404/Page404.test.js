import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { test, expect } from 'jest-without-globals';

import Page404 from './Page404';

test('Render Page 404', () => {
  const component = renderer.create(
    <Page404 />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Error Message', () => {
  const history = createMemoryHistory();
  history.push('/bad-route');

  const { container } = render(
    <Router history={history}>
      <Page404 />
    </Router>,
  );
  expect(container.textContent).toMatch('You are in the Void');
});
