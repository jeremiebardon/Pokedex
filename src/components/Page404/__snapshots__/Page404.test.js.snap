// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Page 404 1`] = `
<div>
  <h3>
    You are in the Void
  </h3>
</div>
`;
